File: backend/src/HealthCore.Domain/Entities/MedicalConsultation.cs
using HealthCore.Domain.Common;

namespace HealthCore.Domain.Entities;

public class MedicalConsultation : BaseEntity
{
    public Guid PatientId { get; set; }
    public Guid DoctorId { get; set; }
    public Guid? AppointmentId { get; set; }

    public string ReasonForVisit { get; set; } = string.Empty;
    public string? Symptoms { get; set; }
    public string? Diagnosis { get; set; }
    public string? Treatment { get; set; }
    public string? Observations { get; set; }
    public string? InternalNotes { get; set; }

    public Patient Patient { get; set; } = null!;
    public Doctor Doctor { get; set; } = null!;
    public Appointment? Appointment { get; set; }
    public ICollection<VitalSign> VitalSigns { get; set; } = new List<VitalSign>();
    public ICollection<Prescription> Prescriptions { get; set; } = new List<Prescription>();
}

File: frontend/src/types/medical-consultation.ts
export interface VitalSign {
  id?: string
  bloodPressure: string | null
  temperature: number | null
  weight: number | null
  height: number | null
  heartRate: number | null
  oxygenSaturation: number | null
}

export interface CreateVitalSignFormData {
  bloodPressure: string
  temperature: number | ""
  weight: number | ""
  height: number | ""
  heartRate: number | ""
  oxygenSaturation: number | ""
}

export interface MedicalConsultation {
  id: string
  patientId: string
  patientName: string
  doctorId: string
  doctorName: string
  appointmentId: string | null
  reasonForVisit: string
  symptoms: string | null
  diagnosis: string | null
  treatment: string | null
  observations: string | null
  internalNotes: string | null
  vitalSigns: VitalSign[]
  createdAt: string
}

export interface CreateMedicalConsultationFormData {
  patientId: string
  doctorId: string
  appointmentId: string | null
  reasonForVisit: string
  symptoms: string
  diagnosis: string
  treatment: string
  observations: string
  internalNotes: string
  vitalSigns: CreateVitalSignFormData
}

export interface UpdateMedicalConsultationFormData {
  doctorId: string
  appointmentId: string | null
  reasonForVisit: string
  symptoms: string
  diagnosis: string
  treatment: string
  observations: string
  internalNotes: string
  vitalSigns: CreateVitalSignFormData
}
